src/lib.rs

```diff
@@ -1,168 +1,236 @@
-    IncludedFileNotFound(PathBuf, Option<PathBuf>),
-    #[error("illegal configuration line '{0}' in '{}'", .1.as_ref().map(|path| path.display().to_string()).unwrap_or(String::from("<unknown>")))]
-    IllegalConfiguration(String, Option<PathBuf>),
-    #[error("failed to read contents of '{0}': {1}")]
-    FailedToReadFile(PathBuf, #[source] std::io::Error),
-}
+    pub fn settings(&self) -> &HashMap<NixConfigKey, NixConfigValue> {
+        &self.settings
+    }
 
-/// Attempt to parse the `nix.conf` at the provided path.
-///
-/// ```rust
-/// # use std::error::Error;
-/// #
-/// # fn main() -> Result<(), Box<dyn Error>> {
-/// std::fs::write(
-///     "nix.conf",
-///     b"experimental-features = flakes nix-command\nwarn-dirty = false\n",
-/// )?;
-///
-/// let nix_conf = nix_config_parser::parse_nix_config_file(&std::path::Path::new("nix.conf"))?;
-///
-/// assert_eq!(
-///     nix_conf.0.get("experimental-features").unwrap(),
-///     "flakes nix-command"
-/// );
-/// assert_eq!(nix_conf.0.get("warn-dirty").unwrap(), "false");
-///
-/// std::fs::remove_file("nix.conf")?;
-/// # Ok(())
-/// # }
-/// ```
-pub fn parse_nix_config_file(path: &Path) -> Result<NixConfig, ParseError> {
-    if !path.exists() {
-        return Err(ParseError::FileNotFound(path.to_owned()));
-    }
-
-    let contents = std::fs::read_to_string(path)
-        .map_err(|e| ParseError::FailedToReadFile(path.to_owned(), e))?;
-
-    self::parse_nix_config_string(contents, Some(path))
-}
+    pub fn settings_mut(&mut self) -> &mut HashMap<NixConfigKey, NixConfigValue> {
+        &mut self.settings
+    }
 
-/// Attempt to parse the `nix.conf` out of the provided [`String`]. The `origin`
-/// parameter is [`Option`]al, and only influences potential error messages.
-///
-/// ```rust
-/// # use std::error::Error;
-/// #
-/// # fn main() -> Result<(), Box<dyn Error>> {
-/// let nix_conf_string = String::from("experimental-features = flakes nix-command");
-/// let nix_conf = nix_config_parser::parse_nix_config_string(nix_conf_string, None)?;
-///
-/// assert_eq!(
-///     nix_conf.0.get("experimental-features").unwrap(),
-///     "flakes nix-command"
-/// );
-/// # Ok(())
-/// # }
-/// ```
-// Mostly a carbon copy of AbstractConfig::applyConfig from Nix:
-// https://github.com/NixOS/nix/blob/0079d2943702a7a7fbdd88c0f9a5ad677c334aa8/src/libutil/config.cc#L80
-// Some things were adjusted to be more idiomatic, as well as to account for the lack of
-// `try { ... } catch (SpecificErrorType &) { }`
-pub fn parse_nix_config_string(
-    contents: String,
-    origin: Option<&Path>,
-) -> Result<NixConfig, ParseError> {
-    let mut settings = NixConfig::new();
-
-    for line in contents.lines() {
-        let mut line = line;
-
-        // skip comments
-        if let Some(pos) = line.find('#') {
-            line = &line[..pos];
-        }
+    pub fn to_settings(&self) -> HashMap<NixConfigKey, NixConfigValue> {
```

Comment:
Since `NixConfig` already implements `Clone` and we offer `into_settings()` I'm not sure this is necessary API but that's fine.